src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -74,8 +74,38 @@ private struct PgoInstrumentationResults
         [DllImport(JitLibrary)]
         private extern static IntPtr jitStartup(IntPtr host);
 
-        [DllImport(JitLibrary)]
-        private extern static IntPtr getJit();
+        private static class JitPointerAccessor
+        {
+            [DllImport(JitLibrary)]
+            private extern static IntPtr getJit();
+
+            public static IntPtr Get()
+            {
+                if (s_jit != IntPtr.Zero)
+                {
+                    return s_jit;
+                }
+
+                lock(typeof(JitPointerAccessor))
+                {
+                    s_jit = getJit();
+                    return s_jit;
+                }
```

Comment:
What thread safety guarantees is the `lock` giving us here? I would expect a double-checked locking pattern at least.